pkg/sbproxy/notifications/handlers/visibilities_handler.go

```diff
@@ -104,11 +108,18 @@ func (vnh *VisibilityResourceNotificationsHandler) OnCreate(ctx context.Context,
 		return
 	}
 
-	platformBrokerName := BrokerProxyName(vnh.VisibilityClient, v.Additional.BrokerName, v.Additional.BrokerID, vnh.ProxyPrefix)
+	platformBrokerName := utils.BrokerProxyName(vnh.VisibilityClient, v.Additional.BrokerName, v.Additional.BrokerID, vnh.ProxyPrefix)
 
-	err := vnh.enableAccessForPlan(ctx, platformBrokerName, v.Additional.ServicePlan.CatalogID, v.Resource.GetLabels())
-	if err != nil {
+	if err := vnh.enableAccessForPlan(ctx, platformBrokerName, v.Additional.ServicePlan.CatalogID, v.Resource.GetLabels()); err != nil {
 		logger.Error(err)
+
+		// if broker is missing, creating it by resync all brokers
+		if _, err := vnh.BrokerClient.GetBrokerByName(ctx, platformBrokerName); err != nil && strings.Contains(strings.ToLower(err.Error()), "not found") {
+			vnh.Resyncer.Resync(ctx, false)
```

Comment:
will "not found" work for both cf & k8s clients?

Reply:
removed